### ISSUE TITLE
Calibrate Probabilities in lgb

### ISSUE THREAD

**Issue by Solly7** [microsoft/LightGBM#2925]
Hi,

I need help in calibrating probabilities in lgb, see my below model code

params = {
    'objective': 'binary',
    'boosting': 'gbdt',
    'max_depth':-1,
    'num_leaves': 31, # 2^max_depth - 1
    'learning_rate': 0.1,
    'min_data_in_leaf':20,
    'min_child_samples': 20,
    'min_child_weight':0.001,
    'verbose': 0,
}

params['metric'] = 'auc'
num_rounds = 5000

lgtrain = lgb.Dataset(train, label=labelTrain)
lgvalid = lgb.Dataset(test, label=labelTest,reference=lgtrain)
dtest = lgb.Dataset(test)

#replace this with cv only
cv_results = lgb.cv(params, 
                    lgtrain, 
                    nfold=10,
                    stratified=False ,
                    num_boost_round = num_rounds,
                    verbose_eval=10,
                    early_stopping_rounds = 10, 
                    seed = 50)

best_nrounds = cv_results.shape[0] - 1
nround = cv_results['auc-mean'].index(np.max(cv_results['auc-mean'])) 
Best_rounds =np.argmax((cv_results['auc-mean']))
best_score = max(cv_results['auc-mean'])
cv_mean = cv_results.iloc[-1, 0]

print('Best number of rounds is: %i' % nround)
print('Best AUC best score is',best_score)

cv_results = pd.DataFrame(cv_results)

lgb_clf = lgb.train(params, 
                    lgtrain, 
                    num_boost_round=best_nrounds ,
                    valid_sets=[lgtrain,lgvalid],
                    early_stopping_rounds=10,
                    verbose_eval=10)

**Comment by StrikerRUS** [comment]
@Solly7 Hey!

LightGBM doesn't have modules for calibrating probabilities. You can try scikit-learn module: https://scikit-learn.org/stable/modules/calibration.html.

I think that StackOverflow is a better place for a such general question.
